### ISSUE TITLE
Creates desktop entry with empty Exec= line

### ISSUE THREAD

**Issue by boredsquirrel** [cosmic-utils/web-apps#51]
1. install quickwebapps from flathub
2. Fedora Kinoite 40 (KDE), Firefox RPM

```
~ ❯❯❯ whereis firefox
firefox: /usr/bin/firefox /usr/lib64/firefox /etc/firefox /usr/etc/firefox /usr/share/man/man1/firefox.1.gz
```

3. create new webapp
  - name: Element
  - URL: https://app.nitro.chat/#/welcome
  - icon: a local one as all downloaded ones are strange
4. create the entry
5. in the list page, there stands "select browser", hovering over it, my cursor has the 🚫 symbol
6. checking the created desktop entry

```
[Desktop Entry]
Version=1.0
Name=Element
Comment=Web App
Exec=
Terminal=false
Type=Application
Icon=/var/home/user/.local/share/icons/QuickWebApps/Element.svg
Categories=GTK;Web;
MimeType=text/html;text/xml;application/xhtml_xml;
StartupWMClass=WebApp-Element2493
StartupNotify=true
X-MultipleArgs=false
X-WebApp-Browser=Select Browser
X-WebApp-URL=https://app.nitro.chat/#/welcome
X-WebApp-Navbar=false
X-WebApp-PrivateWindow=false
X-WebApp-Isolated=false
X-WebApp-CustomParameters=
```

I guess it should have the webappmanager as an exec argument, then launching firefox from there?

Note that for the forseeable future, Firefox Flatpak is very insecure and should not be used. So I need to use FF RPM.

Curious about the development of the app! A big task for sure, I gave up a bit.

**Comment by elevenhsoft** [comment]
Well, the problem with flatpak version of the app is that. We don't have access to /usr directories and all the root stuff. So basically, we can't find what browsers you have installed, since we can't check if this path exist. Flatpak has not access to it.

There is many limitations to it and I never tried to find solution for it also. Probably there is some works around but it require investigation ^^

**Comment by elevenhsoft** [comment]
Also this is kind of bug if it allowed you to create this desktop file.

**Comment by boredsquirrel** [comment]
Hm, I suppose manually entering the path of the browser would be a reasonable workaround. One could probably also use some xdg magic to find the desktop entry of the default browser.

Or have a list with clickable items, where you can select between Flatpak apps and system apps.

The app should for sure check if the browser is empty, and if yes, display such an option. A text input field would work, or such a said list.

I have no development experience really but can help test prereleases.

**Comment by elevenhsoft** [comment]
> I suppose manually entering the path of the browser would be a reasonable workaround

Well, in that case we still don't know which engine browser use so we can't prepare proper exec line. User should also choose if it's Chromium based, Firefox, Falkon, etc. Imho it's too much hassle to support all this.

Flatpak does not allow do it in easy way so we should not look for workarounds because they always will make more problems instead of solving things.

**Comment by boredsquirrel** [comment]
so... should the flatpak be discontinued?

**Comment by elevenhsoft** [comment]
> so... should the flatpak be discontinued?

No I would not say that. Right now I'm kind of lack of ideas and for future work I'm waiting for Cosmic's platform for flatpak, because after recent updates for libcosmic we have to provide Cosmic icons and maybe some other needed dependencies so I'm worry to updating flatpak right now.

**Comment by elevenhsoft** [comment]
Yeah, I wanted to push an update for flatpak but I'm worry I don't know how to bundle cosmic-icons in it :sweat_smile:

**Comment by elevenhsoft** [comment]
Ok, so I'm trying to bundle cosmic-icons but I have some network issues for couple of days and it will be hard task. Anyway, when it's done I will push an update for flathub. Lately was some improvements in the app so it's worth to push update with it :)

**Comment by hepp3n** [comment]
This should not be the case anymore. Soon, there will be an update on flathub.
